csp_adapter_symphony/adapter.py

```diff
@@ -285,22 +302,28 @@ def _send_messages(
     """Read messages from msg_queue and write to symphony. msg_queue to contain instances of SymphonyMessage, or None to shut down"""
     send_message = config.get_retry_decorator()(send_symphony_message)
     message_create_url = config.message_create_url
+    create_im = config.get_retry_decorator()(create_im_stream)
+    im_create_url = config.im_create_url
     header = config.header
 
-    while True:
-        msg = msg_queue.get()
-        msg_queue.task_done()
-        if not msg:  # send None to kill
-            break
-
-        room_id = room_mapper.get_room_id(msg.room)
+    def _send_message(
+        msg: SymphonyMessage,
+    ):
+        if hasattr(msg, "user_id"):  # sending message to a specific user takes precedence over sending to a room
```

Comment:
the convention for an IM would be `room="IM"` so I think to be unambiguous we should continue to do this. So mostly the same as this but first check`room="IM"` and then check `user_id` to determine the user.